src/cluster.c

```diff
@@ -2169,6 +2199,27 @@ int clusterProcessPacket(clusterLink *link) {
             decrRefCount(channel);
             decrRefCount(message);
         }
+    } else if (type == CLUSTERMSG_TYPE_PUBLISHLOCAL) {
```

Comment:
memcpy is done at the first byte of placeholder data array. Unable to use it.

Reply:
I still think we can optimize this a bit, since the code between these two sections is basically the same. It looks like just the pubsub publish command that is called is different.

Reply:
I've merged the two conditional block into one.